### ISSUE TITLE
is_invocable_v is broken

### ISSUE THREAD

**Issue by eugnsp** [boostorg/callable_traits#184]
It seems that `is_invocable_v` variable template is broken. `is_invocable` and `is_invocable_v` have different implementations:
```
template<typename T, typename... Args>
struct is_invocable : detail::is_invocable_impl<T, Args...>::type {
    using type = typename detail::is_invocable_impl<T, Args...>::type;
};

template<typename T, typename... Args>
constexpr bool is_invocable_v =
    detail::is_invocable_impl<detail::traits<T>, Args...>::type::value;
```

With my understanding, it is not `detail::traits<T>` that should be checked for invocability, but simply `T`. As a result, this assertion [fails](https://godbolt.org/z/tzqcB-):

```
#include <boost/callable_traits.hpp>

int test(int) { return 0; }

static_assert(
    boost::callable_traits::is_invocable_v<decltype(test), int> == 
    boost::callable_traits::is_invocable<decltype(test), int>::value
);
```

Originally reported here: https://stackoverflow.com/q/61696567

**Comment by badair** [comment]
Thanks for the bug report, this is gross. Will fix for Boost 1.74 as well as the missing test coverage.

**Comment by badair** [comment]
Fixed via #185
